app/src/test/java/org/oppia/android/app/translation/ActivityLanguageLocaleHandlerTest.kt

```diff
@@ -0,0 +1,264 @@
+
+    val displayLocale = activityLanguageLocaleHandler.displayLocale
+
+    assertThat(displayLocale.localeContext).isNotEqualToDefaultInstance()
+    assertThat(displayLocale.localeContext.languageDefinition.language).isEqualTo(SWAHILI)
+  }
+
+  @Test
+  fun testUpdateLocale_initialized_sameLocale_returnsFalse() {
+    appLanguageLocaleHandler.initializeLocale(computeNewAppLanguageLocale(ENGLISH))
+
+    val isUpdated = activityLanguageLocaleHandler.updateLocale(retrieveAppLanguageLocale())
+
+    // The locale never changed, so there's nothing to update.
+    assertThat(isUpdated).isFalse()
+  }
+
+  @Test
+  fun testUpdateLocale_initialized_differentLocale_returnsTrue() {
+    appLanguageLocaleHandler.initializeLocale(computeNewAppLanguageLocale(ENGLISH))
+
+    val isUpdated =
+      activityLanguageLocaleHandler.updateLocale(computeNewAppLanguageLocale(SWAHILI))
+
+    assertThat(isUpdated).isTrue()
+  }
+
+  @Test
+  fun testUpdateLocale_afterUpdate_newLocale_returnsTrue() {
+    appLanguageLocaleHandler.initializeLocale(computeNewAppLanguageLocale(ENGLISH))
+    activityLanguageLocaleHandler.updateLocale(
+      computeNewAppLanguageLocale(BRAZILIAN_PORTUGUESE)
+    )
+
+    // Change language back.
+    val isUpdated = activityLanguageLocaleHandler.updateLocale(computeNewAppLanguageLocale(SWAHILI))
+
+    // Updating twice with a new locale should lead to an update.
+    assertThat(isUpdated).isTrue()
+  }
+
+  @Test
+  @RunOn(TestPlatform.ROBOLECTRIC, buildEnvironments = [BuildEnvironment.BAZEL])
+  fun testInitializeLocaleForActivity_uninitialized_throwsException() {
+    val configuration = Configuration()
+
+    val exception = assertThrows(IllegalStateException::class) {
+      activityLanguageLocaleHandler.initializeLocaleForActivity(configuration)
+    }
+
+    // The handler can't be updated before it's initialized.
+    assertThat(exception).hasMessageThat().contains("Expected locale to be initialized")
+  }
+
+  @Test
+  fun testInitializeLocaleForActivity_initialized_updatesSystemLocale() {
+    forceDefaultLocale(Locale.ROOT)
+    val configuration = Configuration()
+    appLanguageLocaleHandler.initializeLocale(
+      computeNewAppLanguageLocale(BRAZILIAN_PORTUGUESE)
+    )
+
+    activityLanguageLocaleHandler.initializeLocaleForActivity(configuration)
+
+    // Verify that the system locale has actually changed.
+    assertThat(Locale.getDefault().language).isEqualTo("pt")
+    assertThat(Locale.getDefault().country).isEqualTo("BR")
+  }
+
+  @Test
+  fun testInitializeLocaleForActivity_initialized_updatesConfigurationLocale() {
+    val configuration = Configuration()
+    appLanguageLocaleHandler.initializeLocale(
+      computeNewAppLanguageLocale(BRAZILIAN_PORTUGUESE)
+    )
+
+    activityLanguageLocaleHandler.initializeLocaleForActivity(configuration)
+
+    // Verify that the locale actually changed in the configuration.
+    val locales = configuration.locales
+    assertThat(locales.size()).isEqualTo(1)
```

Comment:
```suggestion
    assertThat(locales).hasSize(1)
```

More Truth-idiomatic.

Reply:
Could not access this API to check directly using hasSize()